addons/mass_mailing/controllers/main.py

```diff
@@ -92,8 +93,11 @@ def unsubscribe(self, mailing_id, opt_in_ids, opt_out_ids, email, res_id, token)
         return 'error'
 
     @http.route('/mail/track/<int:mail_id>/blank.gif', type='http', auth='public')
-    def track_mail_open(self, mail_id, **post):
+    def track_mail_open(self, mail_id, token, **post):
         """ Email tracking. """
+        if not consteq(token, authenticate_message(request.env, 'email-tracking-id-%s-opened' % mail_id)):
```

Comment:
Well for once that's a bit verbose :D . Maybe 
´´´
authenticate_message(request.env, 'mass_mailing-mail_mail-open-%s' % mail_id)
´´´
That way we have a start of convention: module - model - action - ID (form high level to low level) .

What do you think ?

Reply:
Yes, good idea :)